doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+
+In order to achieve above, new process fdbsyncd is added in SwSS. Fdbsyncd would subscribe to FDB entries for a given VLAN once VLAN-VNI mapping is available. Fdbsyncd will maintain the VLAN subscription and send FDB events to Linux FDB.
+
+![EVPN MAC route handling](images/l2macflowcombined.PNG )
+__Figure 8: EVPN MAC route handling__
+
+
+Zebra will install all of the remote MAC routes to kernel. All Remote MAC addresses will be available in VXLAN_FDB_TABLE in APP_DB via Fdbsynd.
+
+```
+VXLAN_FDB_TABLE:{{vlan}}:{{mac_address}}
+    "remote_vtep" : {{ipv4_address}}
+    "type"   : "static" / "dynamic"  ; type of the entry.
+    "vni" : {{vni_id}}
+```
+
+In the case of MAC move from remote to local, remote MAC entry will be removed by Zebra from Linux FDB. Fdbsyncd will remove the remote mac from APP_VXLAN_FDB_TABLE when MAC moves.
+
+#### 4.3.7.1 Local MAC handling
+
+There is no change to fdborch wrt local MAC handling. 
+
+Currently Local MACs learnt by the fdborch are written to the STATE_DB. 
+
+Fdbsyncd listens to the above table and notifies Linux FDB which notifies to Zebra, Zebra to BGP and BGP originates the MAC route. 
+
+The above is applicable for locally added static MACs also. 
+
+![Local MAC Handling](images/localmac.PNG "Figure : Local MAC handling")
+__Figure 9: Local MAC handling__
+
+#### 4.3.7.2 Remote MAC handling
+
+Remote MACs learnt by BGP will be installed in Linux FDB via Zebra; Fdbsyncd will be listening to Linux FDB, MAC learned over Vxlan interface will be populated in APP_VXLAN_FDB_TABLE.
+
+fdborch does the following on an entry add in APP_VXLAN_FDB_TABLE. 
+
+- Store the Remote MAC in an internal cache. The internal cache used for local MACs will be used for storing remote MACs as well. 
+- sai_fdb_entry_add called with the following attributes
+  - SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID with the bridge port id corresponding to the tunnel.
+  - SAI_FDB_ENTRY_ATTR_ENDPOINT_IP with the tunnel DIP.
+  - SAI_FDB_ENTRY_ATTR_TYPE with type SAI_FDB_ENTRY_TYPE_STATIC and SAI_FDB_ENTRY_ATTR_ALLOW_MAC_MOVE as true(corresponding to type dynamic in VXLAN_FDB_TABLE) or false (corresponding to type static in VXLAN_FDB_TABLE)  
+
+![Remote MAC Handling](images/remotemac.PNG "Figure : Remote MAC handling")
+__Figure 10: Remote MAC handling__
+
+
+#### 4.3.7.3 MAC Move handling
+
+##### Local to Local
+
+There is no special handling by SwSS. 
+
+Local to Local MAC moves will be identified by BGP and no MAC route will be re-originated. 
+
+##### Remote to Local
+
+For MAC move cases from remote to local, fdbsyncd removes the 
+remote entry from VXLAN_FDB_TABLE. The fdborch should not delete the 
+MAC from the internal cache and the hardware. 
```

Comment:
For the remote to local case, why the MAC can not delete from the internal cache and the hardware.  Does this mean the MAC address will exist both in remote and local in the hardware?

Reply:
1. The remote to local is initiated by a learn/move  event from the SAI/hardware. 
2. The internal cache would be updated. 
3. The fdbsyncd knows about this when the STATE DB is updated with the MAC which is moved to the local.
4. The VXLAN_FDB_TABLE entry will have to be removed by the fdbsyncd. 
5. After the above step fdborch gets a notification to remove the MAC, the fdborch should identify that this is a MAC which has already been moved and should not delete.  In a normal remote MAC removal, the MAC would have been removed by fdborch from its internal cache.

Reply:
I am still not clear why the remote MAC address can not remove from the hardware. And when the remote MAC address is removed from the hardware. Did you mean the fdborch will keep the remote MAC address in its cache until an event (maybe timeout event?) , Thanks

Reply:
The remote MAC is programmed as not ageable but moveable. So as part of step 1 the device would have re-learnt the MAC over a local port. At this point itself the MAC is moved in the hardware to a local port. The indication is sent to the fdborch and fdbsyncd to update the necessary software state.